src/observerClass.ts

```diff
@@ -20,6 +21,19 @@ export function makeClassComponentObserver(
     componentClass: React.ComponentClass<any, any>
 ): React.ComponentClass<any, any> {
     const target = componentClass.prototype
+
+    if (__DEV__) {
+        if (componentClass[mobxObserverProperty]) {
+            const displayName = getDisplayName(this)
+            console.warn(
+                `The provided component class (${displayName}) 
+                 has already been declared as an observer component.`
+            )
+        } else if (!componentClass[mobxObserverProperty]) {
+            componentClass[mobxObserverProperty] = true
```

Comment:
I'd do the assignment also outside `__DEV__` environments; I can imagine people might somehow attach behavior to it, and get caught unaware in production

Reply:
Okay, cool.  That aligns with my original thought.